File: drills/ts/harnesses/nc18_daily_temperatures.ts
import * as solution from "./solution";

const show = (v: unknown) => JSON.stringify(v) ?? "undefined";

export function run(): [string, string, string][] {
  if (typeof solution.dailyTemperatures !== "function") throw new Error("__signature_mismatch__");
  return [
    [
      "dailyTemperatures([73, 74, 75, 71, 69, 72, 76, 73])",
      show([1, 1, 4, 2, 1, 1, 0, 0]),
      show(solution.dailyTemperatures([73, 74, 75, 71, 69, 72, 76, 73])),
    ],
    ["dailyTemperatures([30, 40, 50, 60])", show([1, 1, 1, 0]), show(solution.dailyTemperatures([30, 40, 50, 60]))],
    ["dailyTemperatures([30, 30, 30])", show([0, 0, 0]), show(solution.dailyTemperatures([30, 30, 30]))],
  ];
}

File: drills/go/harnesses/nc18_daily_temperatures.go
package main

func main() {
	run(func() []testCase {
		return []testCase{
			tc("dailyTemperatures([73,74,75,71,69,72,76,73])", []int{1, 1, 4, 2, 1, 1, 0, 0}, dailyTemperatures([]int{73, 74, 75, 71, 69, 72, 76, 73})),
			tc("dailyTemperatures([30,40,50,60])", []int{1, 1, 1, 0}, dailyTemperatures([]int{30, 40, 50, 60})),
			tc("dailyTemperatures([30,60,90])", []int{1, 1, 0}, dailyTemperatures([]int{30, 60, 90})),
			tc("dailyTemperatures([90])", []int{0}, dailyTemperatures([]int{90})),
		}
	})
}
